env/mock_env.h

```diff
@@ -16,20 +16,51 @@
 #include "env/composite_env_wrapper.h"
 #include "rocksdb/env.h"
 #include "rocksdb/status.h"
+#include "rocksdb/system_clock.h"
 
 namespace ROCKSDB_NAMESPACE {
+// A SystemClock that fakes sleeps.
+class FakeSleepSystemClock : public SystemClockWrapper {
+ public:
+  explicit FakeSleepSystemClock(const std::shared_ptr<SystemClock>& c)
+      : SystemClockWrapper(c), fake_sleep_micros_(0) {}
+
+  void SleepForMicroseconds(int micros) override {
```

Comment:
I may use "FakeSleepForMicroseconds" here and also rename the "MockSleepForMicroseconds" to "FakeSleepForMicroseconds", what do you think. Otherwise, the name itself does not provide the information that we "Fake" the sleep action.

Reply:
The class is FakeSleepSystemClock.  It is meant to override a "Sleep" with a fake.  I am not sure how the renaming does what you want?

Reply:
Right, but after user has something like std<FakeSleepSystemClock> clock, the will just will clock->SleepForMicroseconds, which can be misleading in the code. I may prefer we keep the "Fake" in the function name.

Reply:
This is meant to be a test class that simulates a sleep by just keeping track of the delay and not actually sleeping.  If the internal code calls SleepForMicroseconds and this method is called "FakeSleepForMicroseconds", this method/class will have no effect.  

The point of this class is to NOT do a real Sleep when the code (internal or test) calls Sleep.

The SpecialEnv class does something similar to sometimes (but not always) override the true sleep.